### ISSUE TITLE
Using getTokenAt removes colon symbol followed by a string.

### ISSUE THREAD

**Issue by carlosamaciel** [codemirror/CodeMirror#6302]
We're using the colon symbol to show custom hints for some in-app strings. Example: ":my-term" would do a custom search of 'my-term' from a special list of strings, that we know in advance since the string contains the colon symbol at the start.

But in 5.48.4, the colon symbol gets removed after using getTokenAt. So we end up with "my-term" instead of ":my-term" so we don't know that we have to search from the special list.

Code sample:
```
// User puts ":my-term" in the input and requests hints.
var cursor = editorInstance.getCursor();
var searchString = editorInstance.getTokenAt( cursor );
// searchString is "my-term" instead of ":my-term"
```

**Comment by marijnh** [comment]
Which mode are you using?

**Comment by carlosamaciel** [comment]
It is custom mode if the user puts a special character, otherwise, it is SQL. To add more information on this issue:

1. It was working on a previous version (we used 5.46).
2. We just updated the library to 5.48.4
3. Found out that the getLineTokens gets us the ":" as punctuation string while the term as a separate string.

Our workaround is to use getLineTokens, identify the current search string, and check the previous token to see if it's a colon symbol. Then we send the complete term (colon + search term) to get the custom hints.
